backend/test/modules/email-sending/core/application/SendEmail.mailhog-spec.ts

```diff
@@ -0,0 +1,38 @@
+import { MailhogClient } from 'mailhog-awesome';
+import { NodeMailerEmailSender } from '../../../../../src/modules/email-sending/infrastructure/mailer/NodeMailerEmailSender';
+
+describe('Example email sending using NodeMailer', () => {
+  let mailhog: MailhogClient;
+  const host = 'localhost';
+
+  beforeAll(async () => {
+    mailhog = new MailhogClient({
+      host,
+      port: 8025,
+    });
+  });
+
+  it('check if email arrived on correct email address with correct subject', async () => {
+    //Given
+    const inbox = 'user@test.com';
+    await mailhog.clearInbox(inbox);
+    const emailSender = new NodeMailerEmailSender({ host: host, port: 1025, secure: false });
+
+    //When
+    await emailSender.sendAnEmail({
+      from: 'admin@test.com',
+      to: inbox,
+      subject: 'Test subject',
+      html: 'Test html body',
+    });
+
+    //Then
+    const email = await mailhog.getLastEmail({
+      from: 'admin@test.com',
+      to: inbox,
+      subject: 'Test subject',
+    });
+    expect(email).toBeTruthy();
+    await mailhog.clearInbox(inbox);
```

Comment:
Wiesz co to robi i po co to tutaj? Może będzie na to jakieś lepsze miejsce?

Reply:
> Wiem, że jest tu tylko 1 TC, ale nie lepiej byłoby to wyrzucić do jakieś setupu/teardownu?

proszę już nie wymyślać dobra :-) ? 
1 TC --> ???? 
jakiego setupu, co masz na myśli?

Reply:
> Wiesz co to robi i po co to tutaj? Może będzie na to jakieś lepsze miejsce?

zapewne ma to związek z tym hmmm?
![image](https://user-images.githubusercontent.com/31566345/111044207-7119e900-8447-11eb-8584-e00d22f5d78d.png) 
i tym:
https://bitbar.com/wp-content/uploads/2019/12/parallel-real-device-and-browser-testing.png
Jeśli tak, to obstawiam, że czyścimy tę pocztę, po to aby nie otrzymać mail'a z innego ewentualnego testu?
Czyli w naszym aktualnym przypadku jest to niepotrzebne, ALE jak dojdzie wysyłanie maili w innych modułach za pomocą mailhog'a to już się przyda.... am I right?

Reply:
@nowakprojects jeśli źle myślę to proszę o correctę :-)

Reply:
Dokłanie o to mi chodziło :) Powinieneś sobie przed odpaleniem testu czyli w beforeEach to robić. Upewnić się, że środowisko jest przygotowane do testu. Bo jeśli np. odpalasz sobie takiego mailhoga, wyślesz maila sobie odpalając aplikację, a potem odpalasz test - to już jest jakiś inny mail wysłany. 

1 TC = 1 Test Case. Bo tak to musialbys pamietac po kazdym wykonywac clearInbox. Akurat Ci się udało, bo jest tylko jeden test tutaj :)

Reply:
https://github.com/nowakprojects/CodersCamp2020.Project.FullStack-Node-React.TableSoccerTournaments/pull/37/commits/20b04d802de3614debabc4a381e2dd57733f7b46
takie coś :-) ?